core/admin/mailu/internal/views/dovecot.py

```diff
@@ -20,16 +21,18 @@ def dovecot_passdb_dict(user_email):
         "allow_real_nets": ",".join(allow_nets)
     })
 
+
 @internal.route("/dovecot/userdb/")
 def dovecot_userdb_dict_list():
     return flask.jsonify([
-        user[0] for user in models.User.query.filter(models.User.enabled.is_(True)).with_entities(models.User.email).all()
+        user[0] for user in db.session.execute(db.select(models.User.email).filter_by(enabled=True)).all()
     ])
 
+
 @internal.route("/dovecot/userdb/<path:user_email>")
 def dovecot_userdb_dict(user_email):
     try:
-        quota = models.User.query.filter(models.User.email==user_email).with_entities(models.User.quota_bytes).one_or_none() or flask.abort(404)
+        quota = db.one_or_404(db.select(models.User.quota_bytes).filter_by(email=user_email))
     except sqlalchemy.exc.StatementError as exc:
         flask.abort(404)
     return flask.jsonify({
```

Comment:
Latest CI error is for this function. It is used when someone logs in via imap. 


https://github.com/Mailu/Mailu/actions/runs/6870165755/job/18684540112#step:11:1755
```
 [2023-11-14 22:19:48,356] INFO in nginx: Login attempt for: admin@mailu.io/submission/587 from: 192.168.203.1/36240: success: password
[2023-11-14 22:19:49,176] ERROR in app: Exception on /internal/dovecot/userdb/user@mailu.io [GET]
Traceback (most recent call last):
  File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.11/site-packages/flask_restx/api.py", line 674, in error_router
    return original_handler(e)
           ^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mailu/internal/views/dovecot.py", line 39, in dovecot_userdb_dict
    "quota_rule": f"*:bytes={quota[0]}"
                             ~~~~~^^^
TypeError: 'int' object is not subscriptable
```

Reply:
That query is a relatively "hot path", we should keep the .with_entities() equivalent to (a) reduce the size of what is returned and (b) make it easier on the DBM.

@DjVinnii please ensure you "scope" the selects everywhere we do in the existing code.

Reply:
This is my mistake, I assumed the  `db.select(models.User.quota_bytes)` wouldn't only return the quota_bytes but an object.

According to the SQLAlchemy documentation the `.with_entities()` is part of the legacy query API and replaces the SELECT with the given entities. I think there are 2 way to tackle this:
1. Only select the `quota_bytes` and not the whole model so there is no need to use an index to select the quota bytes.
2. Select the whole model and use `.with_only_columns()` and use the index to target the quota bytes.

For now I went with option 2 in order to keep the changes as close as to the original. I also did the same for other queries with `.with_entities()`